### ISSUE TITLE
Test Pilot wrongly removes also the enabled experiments at uninstall

### ISSUE THREAD

**Issue by SoftVision-PaulOiegas** [mozilla/testpilot#2868]
**[Affected versions]:**
- Firefox 55.0.3 and above

**[Affected Platforms]:**
- All Windows

**[Prerequisites]:**
- The "testpilot.env" string pref was manually added with "stage" value on a new clean profile in "about:config" page.
- The "extensions.webapi.testing" boolean pref was manually added with "true" value on the crated profile.
- Test Pilot stage add-on was already installed on the created profile.

**[Steps to reproduce]:**
1. Open Firefox browser with the profile from prerequisites and navigate to https://testpilot.dev.mozaws.net page.
2. Choose to install Test Pilot add-on.
3. Click on any of the active experiments cards and enable at least one of them
4. Restart the browser.
5. Navigate again to https://testpilot.dev.mozaws.net page and click on "Settings" top right button.
6. While observing the enabled experiment(s) card(s) ,choose to uninstall Test Pilot and confirm the action. 

**[Expected result]:**
- Test Pilot add-on is removed, you are redirected to Homepage but the enabled experiment remains enabled.

**[Actual result]:**
- Test Pilot add-on is removed but you wrongly remain on the experiments page and the enabled experiment is also removed.

**[Notes]:**
- The same behavior occurs even if you manually remove Test Pilot add-on from about:addons page.

**Comment by lmorchard** [comment]
If the Test Pilot add-on was installed from stage and it's 1.1.4, then this is basically a dupe of #2865 - because the old v1.1.4 still has the code to remove experiments on uninstall

**Comment by SoftVision-PaulOiegas** [comment]
Probably, we'll recheck and close it if a fix is landed for #2865.

**Comment by SoftVision-PaulOiegas** [comment]
Sadly this still reproduces after the fix for #2865 has landed. 
This could be easily encountered since it occurs 100% after browser restart. I bet that normal users that install Test Pilot and enable experiments, won't uninstall the TxP add-on in the same session.

**Comment by lmorchard** [comment]
Hmm. I started from a clean profile and installed the add-on from dev, but I couldn't reproduce the issue: Installed Test Pilot, enabled Notes. Restarted. Uninstalled Test Pilot from gear menu, Notes was still installed.

The prerequisites & steps to reproduce seemed a little confusing - specifically the part about "Test Pilot stage add-on was already installed on the created profile". If the add-on from stage is already installed, dev doesn't offer the button to install Test Pilot because it detects the presence of the add-on from stage.

If the old v1 add-on from stage is installed from before the fix in #2865, you'll definitely see experiments uninstalled along with Test Pilot. But, the new v2 add-on doesn't have that implemented.

Wondering if it's that v1 is still in your testing profile, or if there's some other problem here that I'm not seeing

**Comment by SoftVision-PaulOiegas** [comment]
At the moment, the testing was performed on stage, so that's why I've mentioned "stage" for testpilot.env and for add-on. We usually log the issues with the setup for the server on which we are testing at the moment.

However, I have retested the issue on both dev and production servers and it's no longer occurring. This is a bit strange since I was able to reproduce it after you guys fixed the servers builds issues. 
But if I remember well, there was another issue that was not fixed at the time of testing (I think Relud was involved there). Maybe that one was still affecting this flow... 

I will close this one since I'm not able to reproduce it anymore, but I'll keep an eye for any strange behavior in the future. Thanks for the investigations @lmorchard.
